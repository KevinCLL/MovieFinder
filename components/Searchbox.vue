<script>
export default {
  data() {
    return {
      movie: '',
    }
  },
  computed: {
    placeholder() {
      return this.$route.path === '/'
        ? 'Find me a movie'
        : 'Find me another movie'
    },
  },
  methods: {
    handleClick() {
      this.$router.push({
        path: `/find/${this.movie}`,
      })
    },
  },
}
</script>

<template>
  <div class="w-full mb-6 text-center">
    <input
      v-model="movie"
      class="shadow border border-red-500 rounded w-1/2 py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
      :placeholder="placeholder"
      @keyup.enter="handleClick"
    />
    <button
      class="bg-red-500 hover:bg-red-700 text-white font-bold py-2 px-4 rounded"
      :disabled="movie === ''"
      @click="handleClick"
    >
      Go!
    </button>
  </div>
</template>
