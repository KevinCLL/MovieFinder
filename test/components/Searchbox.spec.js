import { mount, createLocalVue } from '@vue/test-utils'
import Searchbox from '@/components/Searchbox.vue'
import VueRouter from 'vue-router'

const localVue = createLocalVue()
localVue.use(VueRouter)

describe('Searchbox', () => {
  let wrapper
  let router

  beforeEach(() => {
    router = new VueRouter()
    wrapper = mount(Searchbox, {
      localVue,
      router,
      data() {
        return {
          movie: 'Batman',
        }
      },
    })
  })
  afterEach(() => {
    if (wrapper) {
      wrapper.destroy()
    }
  })

  test('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  test('placeholder for / is correct', () => {
    expect(wrapper.find('#searchbox').attributes().placeholder).toEqual(
      'Find me a movie'
    )
  })

  test('All other placeholder are correct', () => {
    const batWrapper = mount(Searchbox, {
      mocks: {
        $route: '/find/Batman',
      },
    })
    expect(batWrapper.find('#searchbox').attributes().placeholder).toEqual(
      'Find me another movie'
    )
    batWrapper.destroy()
  })

  test('button is disabled', () => {
    expect(wrapper.find('#searchbutton').attributes('disabled')).toBeFalsy()
    const noMovieWrapper = mount(Searchbox, {
      localVue,
      router,
      data() {
        return {
          movie: '',
        }
      },
    })
    expect(
      noMovieWrapper.find('#searchbutton').attributes('disabled')
    ).toBeTruthy()
    noMovieWrapper.destroy()
  })

  test('should go to Batman page', () => {
    wrapper.find('#searchbutton').trigger('click')
    expect(wrapper.vm.$route.path).toBe('/find/Batman')
  })
})
