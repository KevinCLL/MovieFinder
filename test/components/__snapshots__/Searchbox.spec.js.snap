// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchbox renders correctly 1`] = `
<div
  class="w-full mb-6 text-center"
>
  <input
    class="shadow border border-red-500 rounded w-1/2 py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
    id="searchbox"
    placeholder="Find me a movie"
  />
   
  <button
    class="bg-red-500 hover:bg-red-700 text-white font-bold py-2 px-4 rounded"
    id="searchbutton"
  >
    
    Go!
  
  </button>
</div>
`;
